tests/test_mc_comm.py

```diff
@@ -40,6 +61,54 @@ def test_McCommunicationProcess_super_is_called_during_init(mocker):
     mocked_init.assert_called_once_with(error_queue, logging_level=logging.INFO)
 
 
+@pytest.mark.slow
+@freeze_time("2021-03-16 13:05:55.654321")
+def test_McCommunicationProcess_setup_before_loop__connects_to_boards__and_sends_message_to_main(
+    four_board_mc_comm_process,
+):
+    mc_process = four_board_mc_comm_process["mc_process"]
+    board_queues = four_board_mc_comm_process["board_queues"]
+    assert mc_process.get_board_connections_list() == [None, None, None, None]
+    invoke_process_run_and_check_errors(mc_process, perform_setup_before_loop=True)
+    populated_connections_list = mc_process.get_board_connections_list()
+    assert isinstance(populated_connections_list[0], MantarrayMcSimulator)
+    assert populated_connections_list[1:] == [None, None, None]
+
+    assert_queue_is_eventually_not_empty(board_queues[0][1])
+    process_initiated_msg = board_queues[0][1].get(timeout=QUEUE_CHECK_TIMEOUT_SECONDS)
+    assert process_initiated_msg["communication_type"] == "log"
+    assert (
+        process_initiated_msg["message"]
+        == "Microcontroller Communication Process initiated at 2021-03-16 13:05:55.654321"
+    )
+    # simulator is automatically started by mc_comm during setup_before_loop, so need to stop it
+    populated_connections_list[0].hard_stop()
+
+
+@pytest.mark.slow
+def test_McCommunicationProcess_setup_before_loop__does_not_send_message_to_main_when_setup_comm_is_suppressed():
+    board_queues, error_queue = generate_board_and_error_queues(num_boards=4)
+    mc_process = McCommunicationProcess(
+        board_queues, error_queue, suppress_setup_communication_to_main=True
+    )
+    assert mc_process.get_board_connections_list() == [None, None, None, None]
```

Comment:
for future: it seems like a list of 4 None's is coming up frequently. could maybe extract as a constant at the top of this test module `expected_empty_board_connections_list`